### PR TITLE
Fix Warnings V2

### PR DESCRIPTION
**Pull request by C-SELLERS**
#### Description

Round 2 of warnings cleanups

This time I neatly dealt with them (mostly) by warning, so each commit should reflect a particular warning type.

Following this PR there are only 2 major warnings types, and two singular warnings that I am not entirely sure about.

The major two warning types remaining:
CS0672 - Member overrides obsolete member
CS0618 - Using an obsolete 

For these two I believe we should split these up as there reach will likely be pretty large. I already have one PR for removing the obsolete `SecurityInitializer` from the Universe Selection stack (see #5418) which clears up a decent chunk.

The singular warnings leftover:
CS1998 - async method lacks 'await' operators and will run synchronously; Found in `RealTimeScheduleEventService` Constructor; Not sure about changing this, I think because it is real time its intentional. If so can just ignore.
MSB3270 - Mismatch in processor architecture of the project; related to research architecture. I believe this is intentional so I don't want to break it. 

 

#### Related Issue

#5155 

#### Motivation and Context

Still cleaning up warnings

Down from 1400 before #5408  -> 45 at this point 👍🏽 

#### Requires Documentation Change

N/A

#### How Has This Been Tested?

All unit tests and regressions are green, tested with it rebased on master as well, same results. Can push rebased branch here if need be but there was no merge conflicts anyway.

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

**Comment by C-SELLERS** [comment]
This guy only failed on QuantBook's `DefaultEndDate` test which was fixed by PR #5437 in master

**Comment by C-SELLERS** [comment]
Rebased onto master

**Review by jaredbroad — APPROVED** [review]

File: Indicators/DeMarkerIndicator.cs [hunk 1]
```diff
@@ -59,8 +59,8 @@ public DeMarkerIndicator(int period, MovingAverageType type = MovingAverageType.
         public DeMarkerIndicator(string name, int period, MovingAverageType type = MovingAverageType.Simple)
             : base(name)
         {
-            var _lastHigh = 0m;
-            var _lastLow = 0m;
+            _lastHigh = 0m;
+            _lastLow = 0m;
```
Comment (jaredbroad) [inline Indicators/DeMarkerIndicator.cs:63]: Unrelated for posterity; I think we can remove these; also should be reset in the Reset method.

File: ToolBox/IQFeed/IQFeedDataQueueHandler.cs [hunk 2]
```diff
@@ -57,7 +56,7 @@ public class IQFeedDataQueueHandler : HistoryProviderBase, IDataQueueHandler, ID
         /// <summary>
         /// Gets the total number of data points emitted by this history provider
         /// </summary>
-        public override int DataPointCount => _dataPointCount;
+        public override int DataPointCount { get; } = 0;
```
Comment (jaredbroad) [inline ToolBox/IQFeed/IQFeedDataQueueHandler.cs:59]: Also unrelated posterity; seems unused.

**Review by jaredbroad — APPROVED** [review]

**Pull request merged**